app/pages/resource/reservation-calendar/reservationCalendarSelector.js

```diff
@@ -35,7 +59,43 @@ const timeSlotsSelector = createSelector(
   }
 );
 
+const AvailabilitySelector = createSelector(
+  resourceSelector,
+  (resource) => {
+    if (isEmpty(resource)) {
+      return resource;
+    }
+    const availableTimeByDate = {};
+    resource.openingHours.forEach(({ closes, date, opens }) => {
+      const openMinutes = moment.duration(
+        moment(closes).diff(moment(opens))
+      ).asMinutes();
+      if (availableTimeByDate[date]) {
+        availableTimeByDate[date].openMinutes += openMinutes;
+      } else {
+        availableTimeByDate[date] = {
+          availableMinutes: 0,
+          openMinutes,
+        };
+      }
+    });
+    resource.availableHours.forEach(({ ends, starts }) => {
```

Comment:
- [ ] The availableHours returned from the API won't be correct if reservations have been cancelled. So you have to use the non-cancelled, non-denied reservations for each day to calculate the availability.